internal/provider/resource_tfe_registry_gpg_key.go

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &resourceTFERegistryGPGKey{}
+var _ resource.ResourceWithConfigure = &resourceTFERegistryGPGKey{}
+var _ resource.ResourceWithImportState = &resourceTFERegistryGPGKey{}
```

Comment:
Hey, do you wanna do a static assignability check to [ResourceWithModifyPlan](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework@v1.4.2/resource#ResourceWithModifyPlan) here as well?